### PR TITLE
Add data types for all parameters

### PR DESCRIPTION
**Pull request by smortex**
DATA-TYPE ALL THE THINGS

These changes are supposed to be backward compatible.  I flagged a few parameters which data type is weird, but refrained to change them without approval. Please tell me if you are ok for changing them, and I'll do it!

* Add datatypes to all parameters;
* ~~Improve datatypes flagged with 'FIXME' ;~~ (edit: in another PR, see bellow)
* ~~Remove the 'FIXME's;~~ (edit: in another PR, see bellow)
* ~~Align properly all parameters.~~ (edit: in another PR, see bellow)

### CODE REVIEW

**Comment by smortex** [comment]
@xaque208 sorry to bother you, do you have an opinion regarding changing the types of the parameters marked with *FIXME*?

**Comment by smortex** [comment]
Okay, no problem!  So I guess that this will be a 2 stage process:

* Add data types everywhere (this PR);
* Fix weird types (in a new PR once this one is merged).

I propose to keep the *FIXME* as they are in this PR in order to:

* Not forget one in the future PR;
* If somebody gets annoyed because the data type he used for some parameter is not the expected one and he gets a weird message saying e.g. "Port should be a String, got Integer", seeing the FIXME may maybe make him less upset :-).

The second PR would fix the types, and remove the FIXME.

This PR already have alignment fixes, and the second one will likely be quite small so I may include a commit to align everything, so maybe a third PR will not be necessary, but we will see.

File: manifests/storage/device.pp [hunk 1]
```diff
@@ -18,21 +18,21 @@
...
-  $group           = $bacula::bacula_group,
+  String           $device_name     = $name,
+  String           $media_type      = 'File',
+  String           $device          = '/bacula',
+  Boolean          $label_media     = true,
+  Boolean          $random_access   = true,
+  Boolean          $automatic_mount = true,
+  Boolean          $removable_media = false,
+  Boolean          $always_open     = false,
+  String           $maxconcurjobs   = '1',
```
Comment (xaque208) [inline manifests/storage/device.pp]: Needs a fixme.

**Comment by smortex** [comment]
D'oh! I saw that some files still had no data types 😱 !

I've added a commit to fix that in an easy reviewable way and deployed the change on my infra (no change reported).

**Comment by xaque208** [comment]
Thank you for the efforts @smortex.  I've merged this along with a version requirement bump for stdlib that I needed while testing on my own infrastructure.

**Comment by smortex** [comment]
👍 I deploy using r10k and AFAIK, it does not check those requirements… and I have to admit that for all my internal modules I even do not have a `metdata.json` file and don't really think about this in my workflow.  Do you have some tooling for checking this kind of issues?

Thanks for checking this, I'll continue this work soon!

**Pull request merged**